webpack/assets/javascripts/react_app/pages/common/PageLayout/PageLayout.js

```diff
@@ -61,7 +65,13 @@ const PageLayout = ({
             )}
           </Col>
           <Col id="title_action" md={searchable ? 6 : 8}>
-            <div className="btn-toolbar pull-right">{toolbarButtons}</div>
```

Comment:
Pages no longer have a way of defining their specific buttons - as iiuc any fill added to the slot will affect all pages now

Reply:
The fill will only affect the page in which its loaded on(so adding a `<Fill>` in audits page does not affect statistics page).
So buttons can be added using Fill

Reply:
would the button be removed afterwards when navigating using react router? my understanding of the way slot/fill works is that it will stay on all pages

Reply:
@tbrisker It should stay as long as you keep rendering your `Fills`. If you render a `Fill` in a specific page/route, it will fill the slot only when this page/route active.

IMO we should have a unique `page-id` so we can fill slots in a specific page without the need to implement route/page.

Reply:
why have the slot in the PageLayout at all? if you want to allow plugins to extend a certain page's buttons, add a slot in the relevant page for that. Which again goes back to - moving the logic for which buttons should be displayed to the specific page instead of the PageLayout, and just pass the buttons to the layout like it was before.

Reply:
Assuming the `PageLayout` is actually an `IndexPageLayout` aim to regulate how index pages should look like. In this case, aren't 90% of the pages use export and doc buttons?
If so, should we repeat ourself every time we create an index page?

The repeated code would be:
```js
<Slot multi id="hosts-index-toolbar-button" />
<ExportButton url={url} />
<DocumentationButton url={url} />
```

Reply:
we could create another `IndexPageLayout` component for index pages that specific index pages use instead of the generic `PageLayout`, but still not all pages have all buttons (just go over some of the index pages and let me know how many of them have one or none of these), and still the slot should be defined on the pages we want to allow extending and not all pages. most pages don't have plugins adding buttons to them. more common is a plugin adding actions to a drop-down button (like on the host page). The only exception i think of is the host show page that plugins add buttons to.

Reply:
> we could create another `IndexPageLayout` component for index pages that specific index pages use instead of the generic `PageLayout`

Agree

> but still not all pages have all buttons (just go over some of the index pages and let me know how many of them have one or none of these)

The default atm will be no buttons at all, you will get the doc/export only if you will supply a URL so no one will get them by default.

> and still the slot should be defined on the pages we want to allow extending and not all pages

Why? I thought we do want to have some generic places where we always have a slot.
If I remember correctly the toolbar-buttons was one of the places we identified from the beginning as the best example of such a use case.

> more common is a plugin adding actions to a drop-down button (like on the host page). The only exception i think of is the host show page that plugins add buttons to.

Agree, we will be able to achieve that by wrapping the dropdown items in a Slot.